src/DominantJuice.sol

```diff
@@ -0,0 +1,396 @@
+        delegateAllocations = new JBRedemptionDelegateAllocation3_1_1[](1);
+        delegateAllocations[0] = JBRedemptionDelegateAllocation3_1_1(this, 0, "");
+
+        return (reclaimAmount, memo, delegateAllocations);
+    }
+
+    /// @notice If cycle meets project target, then this function cannot be called, since pauseRedemptions will be on.
+    /// If cycle fails to meet its target, then this function will proceed all the way through for pledgers.
+    function didRedeem(JBDidRedeemData3_1_1 calldata _data) external payable {
+        if (
+            msg.value != 0 || !directory.isTerminalOf(projectId, IJBPaymentTerminal(msg.sender))
+                || _data.projectId != projectId
+        ) revert InvalidRedemptionEvent(msg.sender, _data.projectId, msg.value);
+
+        address payable redeemer = payable(_data.holder);
+        if (pledgedAmount[redeemer] == 0) revert MustBePledger();
+        if (isCycleExpired != true) revert CycleHasNotEndedYet(cycleExpiryDate);
+        if (isTargetMet == true) revert NoRefundsForSuccessfulCycle();
+        if (hasBeenRefunded[redeemer] == true) revert AlreadyWithdrawnRefund();
+
+        // Calculates the refund bonus, based on the time and size of the pledge
+        uint256 pledgerRefundBonus = calculateRefundBonus(redeemer);
+
+        // Sanity Check
+        if (address(this).balance < pledgerRefundBonus) revert InsufficientFunds();
+
+        // This redeeem function is now locked for the current redeeming address.
+        hasBeenRefunded[redeemer] = true;
+
+        // Sending early pledger refund bonus.
+        (bool sendSuccess,) = redeemer.call{value: pledgerRefundBonus}("");
+        require(sendSuccess, "Failed to send refund bonus.");
+        emit CycleRefundBonusWithdrawal(redeemer, pledgerRefundBonus);
+    }
+
+    /**
+     * @dev This function is callable only after cycle expiry if goal is met.
+     * @param receivingAddress Contract Owner must call this function, but they can
+     * input another address to receive funds if desired.
+     */
+    function creatorWithdraw(address payable receivingAddress, uint256 amount) external payable onlyOwner {
+        require(
+            isCycleExpired == true && isTargetMet == true, "Cycle must be expired and successful to call this function."
+        );
+        if (amount > address(this).balance) revert InsufficientFunds();
+
+        (bool success,) = receivingAddress.call{value: amount}("");
+        require(success, "Failed to withdraw cycle funds.");
+
+        emit OwnerWithdrawal(receivingAddress, amount);
+
+        if (address(this).balance == 0) {
+            hasCreatorWithdrawnAllFunds = true;
+        }
+    }
+
+    /// Getters and Helpers
+
+    // Due to decimnal precision and calculating the power of floating-point numbers, we use PRBMath for this helper function.
+    function calculateRefundBonus(address _pledger) public returns (uint256) {
```

Comment:
I read through `DACweights.R` and I think this simplistic "order of pledges" incents front-running. A time-based approach is fairer so people aren't vying for the top spot. The second spot loses 25% of the weight immediately (with the 75% rateOfDecay set above). I understand the incentive to get people to pledge earlier, but you don't want people to game the system or think "well, i'm not going to get anything back now anyway, so what's the point of pledging this 'late' in the game (even though it's the same day)"